doc/src/programmers-guide.md

```diff
@@ -1804,12 +1804,14 @@ Currently, the following options are supported:
 |------------|--------------|-------------|
 | `{socket, reuseaddr}` | `boolean()` | Sets `SO_REUSEADDR` on the socket. |
 | `{socket, linger}` | `#{onoff => boolean(), linger => non_neg_integer()}` | Sets `SO_LINGER` on the socekt. |
+| `{otp, recvbuf}` | `non_neg_integer()` | Sets the buffer size (in bytes) on receive calls (Default value: 512). |
```

Comment:
We should mention it's only supported with platforms using BSD Socket interface (not on Pico).
Also, I think we should raise the default to the MSS value because recvfrom/recv with UDP is supposed to return the first packet and if you have a buffer smaller than this, it is lost, the packet is truncated.

Alternatively, we can consider the default is platform dependent and not have any default in this table.

Reply:
So let's rephrase this and say this is the default buffer size in bytes on receive calls when length is not provided or is 0.

Reply:
So, given what we talked about (this being at the TOP level, is this still the case?  Shouldn't Pico be able to still use this functionality?

Reply:
I also re-worded it to make it clear about the relationship between the default value and the length parameter on the socket:recv functions.